### PR TITLE
fix: refactor deposit logic

### PR DESCRIPTION
**Pull request by ajansari95**
## 1. Overview
- Simplified deposit logic by consolidating common code and eliminating redundancy  wherever possible.

## 2. Implementation details
- Optimized AnyDeposit logic by resolving repetitive if-else blocks.
- Unified deposit types using an enum, removing separate deposit modules

## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: smart-contracts/contracts/cl-vault/src/vault/deposit.rs [hunk 1]
```diff
@@ -84,24 +318,84 @@ pub(crate) fn execute_exact_deposit(
...
+    pool_config: PoolConfig,
+    swap_direction: SwapDirection,
+    token_in_amount: Uint128,
+    swappable_amount: (Uint128, Uint128),
+    deposit_amount_in_ratio: (Uint128, Uint128),
+    max_slippage: Decimal,
+    recipient: &Addr,
+) -> Result<SwapCalculationResult, ContractError> {
+    // TODO check that this math is right with spot price (numerators, denominators) if taken by legacy gamm module instead of poolmanager
+    // TODO check on the twap_window_seconds (taking hardcoded value for now)
```
Comment (arhamchordia) [inline smart-contracts/contracts/cl-vault/src/vault/deposit.rs:337]: Is this todo for this PR?

File: smart-contracts/contracts/cl-vault/src/vault/deposit.rs [hunk 2]
```diff
@@ -303,14 +597,15 @@ mod tests {
         //     )
         //     .unwrap();
```
Comment (arhamchordia) [inline smart-contracts/contracts/cl-vault/src/vault/deposit.rs:598]: Remove commented code.

**Comment by magiodev** [comment]
I don't quite understand why the rename from `any_deposit` to `deposit.rs`. This seems unnecessary and much less consistent. If we want a single `deposit.rs` file, we should also merge in the logic from `exact_deposit.rs`, but I tend to prefer keeping things separate. This is just your preference, but in my opinion, it should be one or the other.

**Review by magiodev — COMMENTED** [review]
I don't quite understand why the rename from `any_deposit` to `deposit.rs`. This seems unnecessary and much less consistent. If we want a single `deposit.rs` file, we should also merge in the logic from `exact_deposit.rs`, but I tend to prefer keeping things separate. This is just your preference, but in my opinion, it should be one or the other.

**Review by magiodev — CHANGES_REQUESTED** [review]
After checking out the branch locally I realized that you did hehe, files changes on Github is not showing any change on exact_deposit.rs file. Anyway, I still think that merging those two in one becomes monolitic and unnecessary.

File: smart-contracts/contracts/cl-vault/src/vault/deposit.rs (hunk 1 above)
Comment (ajansari95) [reply smart-contracts/contracts/cl-vault/src/vault/deposit.rs]: it was added earlier during debugging accounting error, removed.

File: smart-contracts/osmosis/contracts/cl-vault/src/msg.rs [hunk 3]
```diff
@@ -173,6 +173,11 @@ pub struct InstantiateMsg {
     pub initial_upper_tick: i64,
 }
 
+pub enum DepositType {
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/msg.rs:176]: Is this distinction not already made by the two different entrypoints?

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs [hunk 4]
```diff
@@ -1,42 +1,74 @@
...
-    // get the amount of funds we can deposit from this ratio
     let (deposit, refund): ((Uint128, Uint128), (Uint128, Uint128)) =
         get_depositable_tokens(deps.branch(), token0.clone(), token1.clone())?;
 
-    let vault_denom = VAULT_DENOM.load(deps.storage)?;
-    let total_vault_shares: Uint256 = query_total_vault_token_supply(deps.as_ref())?.total.into();
-
-    let user_value = get_asset0_value(deps.storage, &deps.querier, deposit.0, deposit.1)?;
-    let refund_value = get_asset0_value(deps.storage, &deps.querier, refund.0, refund.1)?;
+    // NOTE: both the deposit arms here have some repeated math logic,
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:46]: If we combine both entrypoints into a single function call, why do we still have math logic that is repeated?

**Review by LaurensKubat — CHANGES_REQUESTED** [review]
The addition of the Deposit type enum does not make a whole lot of sense to me since that same information is already implied by the two different entry points. For this PR, I'd remove any changes related to that enum's addition and discuss that in slack and restrict the PR to the math refactor in range.rs

**Pull request closed without merging**